File: cpp/producer-consumer/producer-consumer/Consumer.cpp
#include "Consumer.h"

namespace producer_consumer {

	Consumer::Consumer(AbstractDataQueue* queue)
		: queue(queue)
	{
	}


	Consumer::~Consumer()
	{
	}

	bool Consumer::Consume()
	{
		shared_ptr<ChunkData> data = queue->Dequeue();

		for (auto const& value : data->GetData()) {
			if (IsPrime(value)) {
				return true;
			}
		}
		return false;
	}

	bool Consumer::IsPrime(unsigned int n)
	{
		for (unsigned int i = 1; i < n; i++) {
			if ((n % i) == 0) {
				return false;
			}
		}
		return true;
	}

}  // namespace producer_consumer

File: java/producer-consumer/src/main/java/producer_consumer/Consumer.java
package producer_consumer;

public class Consumer {

    private DataQueue queue;

    public Consumer(DataQueue queue) {
        this.queue = queue;
    }

    public boolean consume() {
        for (int n : queue.dequeue().GetData()) {
            if (isPrime(n)) {
                return true;
            }
        }
        return false;
    }

    boolean isPrime(int n) {
        for (int i = 1; i < n; i++) {
            if ((n % i) == 0) {
                return false;
            }
        }
        return true;
    }
}
